pkg/kubelet/winstats/winstats.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package winstats provides a client to get node and pod level stats on windows
+package winstats
+
+import (
+	"time"
+
+	cadvisorapi "github.com/google/cadvisor/info/v1"
+	cadvisorapiv2 "github.com/google/cadvisor/info/v2"
+)
+
+// Client is an object that is used to get stats information.
+type Client interface {
+	WinContainerInfos() (map[string]cadvisorapiv2.ContainerInfo, error)
+	WinMachineInfo() (*cadvisorapi.MachineInfo, error)
+	WinVersionInfo() (*cadvisorapi.VersionInfo, error)
+}
+
+// StatsClient is a client that implements the Client interface
+type StatsClient struct {
+	winNodeStatsClient
```

Comment:
This embedded `winNodeStatsClient` is unnecessary?

Reply:
This is needed as the `winNodeStatsClient` is passed into `NewClient` (https://github.com/bobbypage/kubernetes/blob/07cfd4676c138c2c45b253392cef70bc7ee2130c/pkg/kubelet/winstats/winstats.go#L61)

Reply:
Doesn't need to explicitly specify this. It will work as long as `StatsClient` implements all functions of the `winNodeStatsClient` interface.

Reply:
I'm a bit confused. My thoughts are
 - `StatsClient` implements the `Client` interface
 -  `StatsClient` uses `winNodeStatsClient`, but it doesn't implement it.
 - `perfCounterNodeStatsClient` implements the `winNodeStatsClient` interface

So `StatsClient` needs to access `winNodeStatsClient`, but doesn't necessarily need to embed it.

Reply:
Makes sense, I fixed this is the last update: https://github.com/bobbypage/kubernetes/blob/7b77b0c10481d4a4d955f55524071b42ee301c7b/pkg/kubelet/winstats/winstats.go#L35

As @yujuhong mentioned, `StatsClient` needs to access `winNodeStatsClient` so there is a reference to it, but it is not embed.